### ISSUE TITLE
hddtemp_smartctl has problem with parsing disk by-id path

### ISSUE THREAD

**Issue by JosefRypacek** [munin-monitoring/munin#1347]
**Describe the bug**
Hi all,
I've configured env.drives to monitor my drives using /dev/disk/by-id/* paths. It works as expected, but one of drives returns error.

```
disk/by-id/ata-WDC_WD40EFRX-68N32N0_WD-WCC7K4ABCDEF # this one works
disk/by-id/ata-WDC_WDS120G2G0B-00EPW0_201278123456 # this one does not work
```

`[ERROR] Command /usr/sbin/smartctl -A --nocheck=standby /dev/disk/by-id/ata-WDC_WDS120G2G0B-00EPW0 on drive disk/by-id/ata-WDC_WDS120G2G0B-00EPW0_201278123456 failed with exit(1): /dev/disk/by-id/ata-WDC_WDS120G2G0B-00EPW0: Unable to detect device type at /etc/munin/plugins/hddtemp_smartctl line 274.`

Problem is probably in parsing. The one drive matches and $1 (path without SN begining with number) is used. Other drives will not match and $drive will be used as $dev.

`my $dev = $drive =~ /(.*)(?:_\d+)$/ ? $1 : $drive;`

**Comment by sumpfralle** [comment]
Thank you for your bug report!

Just for my understanding:

> I've configured env.drives to monitor my drives using /dev/disk/by-id/* paths.

I guess, you just configure a hardcoded list of paths (e.g. `/dev/disk/by-id/...`) for the `drives` environment variable in the plugin configuration. Is this correct?

**Comment by JosefRypacek** [comment]
Yes, it's correct.
I'm trying to use consistent disk paths where possible and hardcoded list of drives is the only way for this plugin without big changes in code, but it's sufficient for me.

**Comment by sumpfralle** [comment]
fixed in cfd803b091ccb8
